include/ProjectVersion.h

```diff
@@ -36,29 +37,27 @@
 class ProjectVersion
 {
 public:
-	enum CompareType { Major, Minor, Release, Stage, Build };
+	enum CompareType { Major=1, Minor=2, Release=3, Stage=4, Build=5 };
 
 	ProjectVersion(QString version, CompareType c = Build);
 	ProjectVersion(const char * version, CompareType c = Build);
 
 	int getMajor() const { return m_major; }
 	int getMinor() const { return m_minor; }
-	int getRelease() const { return m_release; }
-	QString getStage() const { return m_stage; }
-	int getBuild() const { return m_build; }
+	int getPatch() const { return m_patch; }
+	QStringList getLabels() const { return m_labels;}
 	CompareType getCompareType() const { return m_compareType; }
 	ProjectVersion setCompareType(CompareType compareType) { m_compareType = compareType; return * this; }
 
-	static int compare(const ProjectVersion& a, const ProjectVersion& b, CompareType c);
+	static int compare(const ProjectVersion& a, const ProjectVersion& b, int c);
```

Comment:
Couldn't we still leave this as `CompareType`? Might be more verbose using `CompareType::Release`.

Reply:
I wanted to ensure that you could pass in ints other than those in the enum (in other words, >5). From what I can tell, this would require a cast if the parameter type were CompareType, and I'm not sure if it would allow you to pass an int that doesn't correspond to an enum. On the other hand, passing CompareType::Release to a function that expects an int works just fine.

If it's possible to pass ints outside the enum with a cast, I can revert it to CompareType. Otherwise I prefer to leave that option open. (It also occurs to me that I should allow passing -1 to compare the entire version regardless of length)

Reply:
It is possible to cast an `int` (or other numeric type) to an enum type, even when the value don't correspond to any of the enumerators, but some care is required. Converting a value with more bits than the enum requires results in undefined behaviour. The safe approach is to use the type you want to convert from as the underlying type of the enum (e.g. `enum class CompareType : int`), then any value of that type can be safely converted. Otherwise, you need to make sure that any value you convert requires no more bits to store than the largest enumerator.

Reply:
Unresolving. I'll set int as the underlying type for CompareType and change the parameter back. With that done I'll also add `CompareType::All = -1`.

Reply:
Fixed in latest commit. I ended up setting All to `std::numeric_limits<int>::max()`. With `-1`, CompareTypes weren't correctly ordered, which broke the "use the smaller CompareType when comparing" logic. I also made All the new default comparison, since I think that's the desired behavior.